### ISSUE TITLE
Tests in removal and online_offline fail due to 'no device in pool'

### ISSUE THREAD

**Issue by ZackaryWelch** [zfsonfreebsd/ZoF#39]
Almost all removal tests fail due to:
```
cannot remove md0: no such device in pool
ERROR: zpool remove testpool md0 exited 1
```
In removal_all_vdev, the error occurs first with md1. In removal_multiple_indirection, removal_with_errors, and removal_resume_export, the tests fail beforehand due to a zinject bug. In removal_remap_deadlists, there is an error with zdp. In all the error occurs in 17 tests in removal, though could possible include those that fail beforehand (additional 4).

These errors also occur in online_offline_001 and online_offline_002 with:
```
cannot offline md0: no such device in pool
```
Edit:
See also cli_root/zpool_offline/zpool_offline_001, zpool_offline_003, and zpool_online/zpool_online_001 for the same error. Additionally pool_checkpoint/checkpoint_removal. This also occurs in detachment with zpool_scrub/zpool_scrub_005 as well as zpool_resilver/zpool_resilver_restart with:
```
cannot detach md1: no such device in pool
ERROR: zpool detach testpool md1 exited 255
```
This may be an issue with how zpool detects paths in FreeBSD.

**Comment by kmoore134** [comment]
@mattmacy Ok, found the issue here:

```
% sudo ./zpool status
  pool: testpool
 state: ONLINE
  scan: none requested
config:

        NAME        STATE     READ WRITE CKSUM
        testpool    ONLINE       0     0     0
          mirror-0  ONLINE       0     0     0
            md0     ONLINE       0     0     0
            md1     ONLINE       0     0     0
            md2     ONLINE       0     0     0

errors: No known data errors
% sudo ./zpool detach testpool md1
cannot detach md1: no such device in pool
% sudo ./zpool detach testpool /dev/md1
```

Expected behavior is that you can interact with devices without needing the "/dev"

**Comment by mattmacy** [comment]
@kmoore134  8eab9ee0abd30a0db5eb42d4ee8ec608c7456cae please test

**Comment by ZackaryWelch** [comment]
Failed testing.

**Comment by mattmacy** [comment]
needs repro
```
mmacy@beastie3 [~|18:20|6] doas zpool create testpool mirror md0 md1 md2
mmacy@beastie3 [~|18:20|7] doas zpool detach testpool md1              
mmacy@beastie3 [~|18:20|8] 
```

**Comment by ZackaryWelch** [comment]
Apologies, what I was testing was online_offline. Those still fail unless '/dev/' is put, then they pass. Try `zpool status -v` then run `zpool offline md0`. That will fail unless you run `zpool offline /dev/md0`.

**Comment by mattmacy** [comment]
fixed by 81493c35f92e421e3099dc8e83fb0131d0caaf49
